src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,888 @@
+			}
+			else if (hasIconMenuItem)
+			{
+				VisualStateManager.GoToState(this, "IconPlaceholder", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "NoPlaceholder", bUseTransitions);
+			}
+
+			// PaddingSizeStates
+			if (shouldBeNarrow)
+			{
+				VisualStateManager.GoToState(this, "NarrowPadding", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "DefaultPadding", bUseTransitions);
+			}
+
+
+		}
+
+		// MenuFlyoutSubItem's presenter size changed event handler that
+		// adjust the sub presenter position to the proper space area
+		// on the available window rect.
+		void OnPresenterSizeChanged(
+		object pSender,
+		SizeChangedEventArgs args)
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnPresenterSizeChanged.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnPresenterSizeChanged(pSender, args, m_tpPopup as Popup);
+
+			if (m_tpMenuPopupThemeTransition == null)
+			{
+
+				MenuFlyoutPresenter parentMenuFlyoutPresenter;
+				parentMenuFlyoutPresenter = GetParentMenuFlyoutPresenter();
+
+				// Get how many sub menus deep we are. We need this number to know what kind of Z
+				// offset to use for displaying elevation. The menus aren't parented in the visual
+				// hierarchy so that has to be applied with an additional transform.
+				int depth = 1;
+				if (parentMenuFlyoutPresenter != null)
+				{
+					depth = parentMenuFlyoutPresenter.GetDepth() + 1;
+				}
+
+				// UNO TODO
+				//Transition spMenuPopupChildTransition;
+				//(MenuFlyout.PreparePopupThemeTransitionsAndShadows((Popup*)(m_tpPopup), 0.67 /* closedRatioConstant */, depth, &spMenuPopupChildTransition));
+				//(spMenuPopupChildTransition as MenuPopupThemeTransition.Direction = xaml_primitives.AnimationDirection_Top);
+				//m_tpMenuPopupThemeTransition = spMenuPopupChildTransition;
+			}
+
+			// Update the OpenedLength property of the ThemeTransition.
+			double openedLength = (m_tpPresenter as Control).ActualHeight;
+
+			// UNO TODO
+			// (m_tpMenuPopupThemeTransition as MenuPopupThemeTransition).OpenedLength = openedLength;
+		}
+
+		void ClearStateFlags()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: ClearStateFlags.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.ClearStateFlags();
+
+		}
+
+		void OnIsEnabledChanged(/*IsEnabledChangedEventArgs* args*/)
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnIsEnabledChanged.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnIsEnabledChanged();
+		}
+
+		void OnVisibilityChanged()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnVisibilityChanged.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnVisibilityChanged();
+
+		}
+
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			//*ppAutomationPeer = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)